lib/ansible/plugins/lookup/__init__.py

```diff
@@ -101,3 +102,19 @@ def run(self, terms, variables=None, **kwargs):
             result_string = to_unicode(result_string)
         """
         pass
+
+    def find_needle(self, myvars, subdir, needle):
```

Comment:
And/or, throw some variety of IGotTooManyAnswersException if find_file_in_search_paths end up with len(result) > 1

something like:
``` python
def find_file_in_search_paths(self, myvars, subdirs, target_name):
    results = []
    for subdir in subdirs:
          try:
               result.append(self.find_file_in_search_path(myvars, subdir, target_name))
          # could use a more specific exception here. Or let find_needle/find_file_in_search_path() return results as []
         except AnsibleError:
              continue
   if  not results:
       raise NoFilesFoundInSearchPathException(target_file=target_name, subdir=subdir)
 
  if len(results) > 1:
      raise TooManyFilesException(target_file=target_name, subdir=subdir, results=results)
   return results
```